src/Chainweb/Pact/PactService.hs

```diff
@@ -186,35 +187,35 @@ initialPayloadState
 initialPayloadState Test{} _ _ = pure ()
 initialPayloadState TimedConsensus{} _ _ = pure ()
 initialPayloadState PowConsensus{} _ _ = pure ()
-initialPayloadState v@TimedCPM{} cid mpa = initializeCoinContract v cid mpa
-initialPayloadState v@Development cid mpa = initializeCoinContract v cid mpa
-initialPayloadState v@Testnet00 cid mpa = initializeCoinContract v cid mpa
-initialPayloadState v@Testnet01 cid mpa = initializeCoinContract v cid mpa
-initialPayloadState v@Testnet02 cid mpa = initializeCoinContract v cid mpa
+initialPayloadState v@TimedCPM{} cid mpa = initializeCoinContract v cid mpa TN02.payloadBlock
+initialPayloadState v@Development cid mpa = initializeCoinContract v cid mpa D.payloadBlock
+initialPayloadState v@Testnet02 cid mpa = initializeCoinContract v cid mpa TN02.payloadBlock
 
 initializeCoinContract
-    :: PayloadCas cas
+    :: forall cas. PayloadCas cas
     => ChainwebVersion
     -> ChainId
     -> MemPoolAccess
+    -> PayloadWithOutputs
     -> PactServiceM cas ()
-initializeCoinContract v cid mpa = do
+initializeCoinContract v cid mpa pwo = do
     cp <- view (psCheckpointEnv . cpeCheckpointer)
     genesisExists <- liftIO $ lookupBlockInCheckpointer cp (0, ghash)
     when (not genesisExists) createCoinContract
 
   where
+    ghash :: BlockHash
     ghash = _blockHash genesisHeader
+
+    createCoinContract :: ReaderT (PactServiceEnv cas) (StateT PactServiceState IO) ()
     createCoinContract = do
         txs <- execValidateBlock mpa genesisHeader inputPayloadData
         bitraverse_ throwM pure $ validateHashes txs genesisHeader
 
-    PayloadWithOutputs{..} = payloadBlock
```

Comment:
Bug: Pact Service was being initialized with `Testnet00`'s payload, regardless of `ChainwebVersion`. We have no idea how the system worked despite this.

@emilypi @emmanueldenloye @marklnichols any thoughts?